### PR TITLE
O3-2242: 'Retrospective Entry' label on Chart Header when doing RDE

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Puts a label in the top bar when in retrospective entry mode. Makes the patient banner visit tag only reflect whether there is an ongoing visit, so it has nothing to do with RDE.

## Screenshots
Screencast from 2023-07-25 20-09-28

## Related Issue
https://issues.openmrs.org/browse/O3-2242

## Other
Pending https://github.com/openmrs/openmrs-esm-core/pull/738

### CODE REVIEW

**Comment by brandones** [comment]
@ibacher @denniskigen ?

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx [hunk 1]
```diff
@@ -91,17 +82,21 @@ const PatientInfo: React.FC<PatientInfoProps> = ({ patient }) => {
...
+      {patientNameIsTooLong ? (
+        <Tooltip
+          align="bottom-left"
+          width={100}
+          label={
+            <>
+              <p className={styles.tooltipPatientName}>{name}</p>
+              <p className={styles.tooltipPatientInfo}>{`${parseInt(age(patient?.birthDate))}, ${getGender(
+                patient?.gender,
+              )}`}</p>
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:96]: I'm a bit unclear on why this extra information is included here but not, say, as a tooltip on the patient name if it's 25 characters or less.

**Review by ibacher — APPROVED** [review]
So, I think we'd still want a retrospective entry tag in addition to the header display. I know that kind of duplicates information on the screen, but it's quite important context (since a user might get used to just looking for the Active Visit tab).

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx (hunk 1 above)
Comment (brandones) [reply packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:96]: Yeah that is goofy. I'll fix it. Oh yeah sorry I forgot to mention this PR includes a bunch of random refactoring.

**Comment by brandones** [comment]
> So, I think we'd still want a retrospective entry tag in addition to the header display. I know that kind of duplicates information on the screen, but it's quite important context (since a user might get used to just looking for the Active Visit tab).

A lot is in the air on the design side right now. For now I'm gonna merge this in, and then we'll see where the design lands.

File: packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx (hunk 1 above)
Comment (brandones) [reply packages/esm-patient-chart-app/src/visit-header/visit-header.component.tsx:96]: Actually looking at it now, I think it makes sense. My only concern is that the patient name by default looks kinda clickable, when it is actually only clickable if it is a very long name.

image

image

**Pull request merged**